src/smartAsset.js

```diff
@@ -182,7 +182,8 @@ export default (initialOptions = {}) => {
         return
       }
 
-      const id = importer ? join(dirname(importer), source) : source
+      const createId = (importer, source) => isAbsolute(source) ? source : join(dirname(importer), source)
+      const id = importer ? createId(importer, source) : source
```

Comment:
I suggest to simplify code to:

```
const id = importer && !isAbsolute(source) ? join(dirname(importer), source) : source
```

Please also add a unit test.

Reply:
Added tests